Mathlib/CategoryTheory/Abelian/GrothendieckCategory.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 Paul Reichert. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+
+import Mathlib.CategoryTheory.Abelian.GrothendieckAxioms.Basic
+import Mathlib.CategoryTheory.Abelian.Subobject
+import Mathlib.CategoryTheory.Abelian.Transfer
+import Mathlib.CategoryTheory.Adjunction.AdjointFunctorTheorems
+import Mathlib.CategoryTheory.Limits.HasLimits
+
+/-!
+
+# Grothendieck categories
+
+This file defines Grothendieck categories and proves basic facts about them.
+
+## Definitions
+
+A `GrothendieckCategory` is an abelian category provided that it has `AB5` and a separator.
+
+## Theorems
+
+Relevant implications of `GrothendieckCategory` are established in `GrothendieckCategory.hasLimits`
+and `GrothendieckCategory.hasColimits`.
+
+## References
+
+* [Stacks: Grothendieck's AB conditions](https://stacks.math.columbia.edu/tag/079A)
+
+-/
+
+namespace CategoryTheory
+
+open Limits
+
+universe w v u w₂ v₂ u₂
+variable (C : Type u) [Category.{v} C] (D : Type u₂) [Category.{v₂} D]
+
+/--
+In the literature, an abelian category `C` is called a Grothendieck category provided that it has
+`AB5` and a separator (see `HasSeparator`).
+
+`IsGrothendieckAbelian.{w} C` is defined such that it holds if and only if `C` is equivalent to a
+Grothendieck category -- more concretely, if and only if `ShrinkHoms.{w} C` is a Grothendieck
+category.
```

Comment:
If `C` is an abelian category, we shall say that it satisfies `IsGrothendieckAbelian.{w} C`
if it is locally small (relative to `w`), has exact filtered colimits of size `w` (AB5) and has a separator.
If `[Category.{v} C]` and `w = v`, this means that `C` satisfies AB5 and has a separator;
general results about Grothendieck abelian categories can be
reduced to this case using the instance `ShrinkHoms.isGrothendieckAbelian` below.
The introduction of the auxiliary universe `w` shall be needed for certain
applications to categories of sheaves.